### ISSUE TITLE
3 minutes to start Cypress + Meteor (using caches)

### ISSUE THREAD

**Issue by haveaguess** [cypress-io/github-action#435]
I know you must be busy but perhaps you can help.

Feels like there must be something wrong? 

Here's the log, it seems to be using the github, cypress, NPM caches?

```
Run cypress-io/github-action@v2
  with:
    install-command: meteor npm i
    start: meteor npm run start
    wait-on: http://localhost:3000
    wait-on-timeout: 240
    parallel: false
    spec: tests/cypress/integration/meetingFeedback.spec.js
  tests/cypress/integration/allTests.spec.js
  tests/cypress/integration/teamPods.spec.js
  tests/cypress/integration/autoOrganizationSignup.spec.js
  tests/cypress/integration/podPersonal.spec.js
  tests/cypress/integration/meeting.spec.js
  tests/cypress/integration/agendaEmailSend.spec.js
  
    record: false
    config-file: cypress.json
Received 0 of 176796132 (0.0%), 0.0 MBs/sec
Received 0 of 44405832 (0.0%), 0.0 MBs/sec
Received 44405832 of 44405832 (100.0%), 23.3 MBs/sec
Cache Size: ~42 MB (44405832 B)
/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/03724d9e-2ff5-4d6a-8200-f60aaac4b268/cache.tzst -P -C /home/runner/work/connect4/connect4
Received 62914560 of 176796132 (35.6%), 29.9 MBs/sec
Cache restored successfully
Received 150994944 of 176796132 (85.4%), 47.9 MBs/sec
Received 176796132 of 176796132 (100.0%), 44.1 MBs/sec
Cache Size: ~169 MB (176796132 B)
/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/73e037e0-b0e9-4caf-a12e-85dfa596a328/cache.tzst -P -C /home/runner/work/connect4/connect4
Cache restored successfully
install command command "meteor npm i"
current working directory "/home/runner/work/connect4/connect4"
/usr/local/bin/meteor npm i

> bcrypt@5.0.1 install /home/runner/work/connect4/connect4/node_modules/bcrypt
> node-pre-gyp install --fallback-to-build

[bcrypt] Success: "/home/runner/work/connect4/connect4/node_modules/bcrypt/lib/binding/napi-v3/bcrypt_lib.node" is installed via remote

> bufferutil@4.0.3 install /home/runner/work/connect4/connect4/node_modules/bufferutil
> node-gyp-build


> mongodb-client-encryption@1.2.6 install /home/runner/work/connect4/connect4/node_modules/mongodb-client-encryption
> prebuild-install --tag-prefix node-v || node-gyp rebuild


> puppeteer@8.0.0 install /home/runner/work/connect4/connect4/node_modules/puppeteer
> node install.js


Chromium (856583) downloaded to /home/runner/work/connect4/connect4/node_modules/puppeteer/.local-chromium/linux-856583

> utf-8-validate@5.0.5 install /home/runner/work/connect4/connect4/node_modules/utf-8-validate
> node-gyp-build


> cypress@8.4.1 postinstall /home/runner/work/connect4/connect4/node_modules/cypress
> node index.js --exec install

Note: Overriding Cypress cache directory to: /home/runner/.cache/Cypress

      Previous installs of Cypress may not be found.


Cypress 8.4.1 is installed in /home/runner/.cache/Cypress/8.4.1

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 1180 packages from 838 contributors and audited 1185 packages in 29.188s

126 packages are looking for funding
  run `npm fund` for details

found 47 vulnerabilities (35 moderate, 4 high, 8 critical)
  run `npm audit fix` to fix them, or `npm audit` for details
/usr/local/bin/npx cypress cache list
┌─────────┬───────────────────┐
│ version │ last used         │
├─────────┼───────────────────┤
│ 8.4.1   │ a few seconds ago │
└─────────┴───────────────────┘
start server "meteor npm run start command "meteor npm run start"
current working directory "/home/runner/work/connect4/connect4"
waiting on "http://localhost:3000" with timeout of 240 seconds
/usr/local/bin/meteor npm run start

> connect4@ prestart /home/runner/work/connect4/connect4
> meteor npm install; meteor npm run lint

npm WARN @material-ui/data-grid@4.0.0-alpha.37 requires a peer of react@^17.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @material-ui/utils@5.0.0-beta.4 requires a peer of react@^17.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN cypress-browser-permissions@1.0.5 requires a peer of cypress@4.x but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

audited 1185 packages in 5.952s

126 packages are looking for funding
  run `npm fund` for details

found 47 vulnerabilities (35 moderate, 4 high, 8 critical)
  run `npm audit fix` to fix them, or `npm audit` for details

> connect4@ lint /home/runner/work/connect4/connect4
> eslint . --ext .jsx --ext .style.js


> connect4@ start /home/runner/work/connect4/connect4
> meteor --settings settings.json --exclude-archs web.browser.legacy,web.cordova

[[[[[ ~/work/connect4/connect4 ]]]]]

=> Started proxy.
=> Meteor 2.4 is available. Update this project with 'meteor update'.
=> Started MongoDB.
=> Started your app.

=> App running at: http://localhost:3000/
```

**Comment by haveaguess** [comment]
> @haveaguess any pointers what is wrong here?

Hi Dmitry, 

Sorry I just wondered if 3 minutes is normal for startup time, seems quite long even with all that caching? What do you think?

If you think it's worth investigating/agree, any idea how I can profile / get more diagnostics on timing?

Thanks for you reply 
Ben

**Comment by MikeMcC399** [comment]
@haveaguess 

This is quite an old open issue.

From what I could find out, the npm [meteor](https://www.npmjs.com/package/meteor) package states
> Node.js <=14.x and npm <=6.x is recommended.

and these versions are no longer supported. Node.js `18` is the earliest supported version.

I'm closing this issue since it is stale and probably no longer relevant. If there is any new information regarding an issue about usage with current versions of Cypress we can re-open if appropriate.
